### ISSUE TITLE
Pyodide doesn't work when linked with `-sASSERTIONS`

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#3861]
## 🐛 Bug

It fails with 

```
Aborted(Assertion failed: native function `py_version_major` called before runtime initialization)
```

### To Reproduce

Link Pyodide with `-sASSERTIONS` then try to load it.

**Comment by ryanking13** [comment]
Hmm, perhaps we should delay filesystem initialization to the `preMain` stage? Not sure it will work though.

**Comment by hoodmane** [comment]
Yeah this stuff is really confusing tbh. It will probably take a bit of experimentation to fix this. In the meantime, when you compile with assertions you can just delete this one.
